twisted/conch/ssh/keys.py

```diff
@@ -802,9 +808,13 @@ def fingerprint(self):
 
         @rtype: L{str}
         """
-        return nativeString(
-            b':'.join([binascii.hexlify(x)
-                       for x in iterbytes(md5(self.blob()).digest())]))
+        fingerprint_digest=hash_function(self.blob()).digest()
+        if hash_function==sha256:
+            return nativeString(base64.b64encode(fingerprint_digest))
+        else:
+            return nativeString(
+                b':'.join([binascii.hexlify(x)
+                           for x in iterbytes(fingerprint_digest)]))
```

Comment:
And in fact, @adiroiban has noted in the ticket itself that we may want to be able to control the representation as well. That would mean we'd want another enum (probably using `twisted.python.constants.Names`, not `Values`) that controls the representation.